cc_core.c

```diff
@@ -538,6 +538,75 @@ void emit_move(int destination_reg, int source_reg, char* note)
 	}
 }
 
+void emit_load_relative_to_register(int destination, int offset_register, int value, char* note)
+{
+	char* destination_name = register_from_string(destination);
+	char* offset_name = register_from_string(offset_register);
+	char* value_string = int2str(value, 10, TRUE);
+
+	if((KNIGHT_POSIX == Architecture) || (KNIGHT_NATIVE == Architecture))
+	{
+		emit_out("ADDI R");
+		emit_out(destination_name);
+		emit_out(" R");
+		emit_out(offset_name);
+		emit_out(" ");
+		emit_out(value_string);
+	}
+	else if((X86 == Architecture) || (AMD64 == Architecture))
+	{
+		emit_out("lea_");
+		emit_out(destination_name);
+		emit_out(",[");
+		emit_out(offset_name);
+		emit_out("+DWORD] %");
+		emit_out(value_string);
+	}
+	else if(ARMV7L == Architecture)
+	{
+		emit_out("!");
+		emit_out(value_string);
+		emit_out(" ");
+		emit_out(destination_name);
+		emit_out(" SUB ");
+		emit_out(offset_name);
+		emit_out(" ARITH_ALWAYS\n");
+	}
+	else if(AARCH64 == Architecture)
+	{
+		/* TODO: We use register one as a temporary which might be surprising. */
+		emit_move(destination, offset_register, note);
+		emit_load_immediate(REGISTER_ONE, value, note);
+		emit_out("SUB_");
+		emit_out(destination_name);
+		emit_out("_");
+		emit_out(destination_name);
+		emit_out("_X1");
+	}
+	else if((RISCV32 == Architecture) || (RISCV64 == Architecture))
+	{
+		emit_out("rd_");
+		emit_out(destination_name);
+		emit_out(" rs1_");
+		emit_out(offset_name);
+		emit_out(" !");
+		emit_out(value_string);
+		emit_out(" addi");
```

Comment:
This still suffers from https://github.com/oriansj/M2-Planet/issues/77

Probably unlikely to hit it in practice but we are limited to 11-bit offsets here.

Reply:
Oh it is fixed in later commits...